### DISCUSSION TITLE
How to resize a image or path object?

### DISCUSSION THREAD

**Discussion by jdschwam** [spakin/SimpInkScr#136] [Q&A] [answered]
I enjoy SIS.  Thank you.  It is doing everything I need, but there is one feature I have yet to get my head around.  How to resize a path object that was imported via objects_from_svg_file().  I've tried for a month, learned a lot, and re-size is a no-go.  I've mostly focused around Transform. 

What I need to do is change variable Height/Width objects of various canvas sizes and viewport scales (embedded and not), into fixed sized (25mm x 25mm) objects.  

This is an example of what I am trying.... the transform correctly positions the XY origin, but resizing does not

 # Modify Box
def get_percentage_increase(num_a, num_b):
        try:
            return abs((num_a - num_b) / num_b) * 100
       except ZeroDivisionError:
            return float('inf')

    hc_scale = get_percentage_increase(25, horse_cow_obj.bounding_box().width)

    # x0 = horse_cow_obj._transform.matrix[0][2]
    # y0 = horse_cow_obj._transform.matrix[1][2]
    hc_x = 5
    hx_y = 5

    try:
        horse_cow_obj.scale(hc_scale, 'ul')
        horse_cow_obj.transform ='translate(%s, %s), first= True' % (hc_x, hc_y)
        print("Horse-Cow Mod successful")
    except:
        print("Horse-Cow Mod failed")

Any help is greatly appreciated.  Thank you, Dan

**Comment by spakin** [comment, marked as answer]
I see two mistakes in the preceding code.  First,
```Python
hx_y = 5
```
should be
```Python
hc_y = 5
```
Second, in
```Python
horse_cow_obj.transform ='translate(%s, %s), first= True' % (hc_x, hc_y)
```
the [`transform` method](https://github.com/spakin/SimpInkScr/wiki/Modifying-existing-objects#replacing-the-entire-transformation) accepts either an [`inkex.Transform`](https://inkscape.gitlab.io/extensions/documentation/source/inkex.transforms.html#inkex.transforms.Transform) or a string that can be used to construct an `inkex.Transform`.  Neither of those recognize `first=True`.  You were probably thinking of
```Python
horse_cow_obj.translate((hc_x, hc_y), first=True)
```

If you're targeting 25mm × 25mm objects, see [Unit conversions](https://github.com/spakin/SimpInkScr/wiki/Other-features#unit-conversions) in the Simple Inkscape Scripting documentation and also the documentation for Inkscape's [`inkex.units` module](https://inkscape.gitlab.io/extensions/documentation/source/inkex.units.html).  Those may facilitate targeting an absolute size.

**Reply by jdschwam** [reply]
Scott, Thanks for your quick reply.  As you can guess, python coding is still new to me.
Thank you so much for the directions to pursue.  I have looked into units before, for general reasons, but this topic is still confusing to me.  With your direction, I will take on the effort to become very familiar with them.  Thank you. Have a great Holiday weekend!  Dan
